frontend/src/metabase/setup/tests/common.unit.spec.tsx

```diff
@@ -164,20 +126,75 @@ describe("setup (OSS)", () => {
         expectSectionToHaveLabel("Add your data", "4");
         expectSectionToHaveLabel("Usage data preferences", "5");
       });
+    });
+  });
 
-      it("should not set the flag for the embedding homepage", async () => {
-        await setupForUsageQuestion();
-        selectUsageReason("self-service-analytics");
-        clickNextStep();
+  describe("embedding homepage flags", () => {
+    it("should set the correct flags when interested in embedding", async () => {
+      await setup();
+      skipWelcomeScreen();
+      skipLanguageStep();
+      await submitUserInfoStep();
+
+      selectUsageReason("embedding");
+      clickNextStep();
+
+      screen.getByText("Finish").click();
+
+      expect(await getLastSettingsPutPayload()).toMatchObject({
+        "embedding-homepage": "visible",
+        "enable-embedding": true,
+        "setup-embedding-autoenabled": true,
+      });
+    });
+
+    it("should not set 'embedding-homepage' when not interested in embedding", async () => {
+      await setup();
+      skipWelcomeScreen();
+      skipLanguageStep();
+      await submitUserInfoStep();
+
+      selectUsageReason("self-service-analytics");
+      clickNextStep();
+
+      screen.getByText("I'll add my data later").click();
+
+      screen.getByText("Finish").click();
+
+      const flags = await getLastSettingsPutPayload();
+
+      expect(flags["embedding-homepage"]).toBeUndefined();
+      expect(flags["enable-embedding"]).toBeUndefined();
+      expect(flags["setup-embedding-autoenabled"]).toBeUndefined();
+    });
 
-        screen.getByText("I'll add my data later").click();
+    it("should not autoenable embedding if it was set by an env", async () => {
+      await setup({
+        settingOverrides: [
+          createMockSettingDefinition({
+            key: "enable-embedding",
+            value: false,
+            is_env_setting: true,
+          }),
+        ],
+      });
+      skipWelcomeScreen();
+      skipLanguageStep();
+      await submitUserInfoStep();
+
+      selectUsageReason("embedding");
+      clickNextStep();
 
-        screen.getByRole("button", { name: "Finish" }).click();
+      screen.getByText("Finish").click();
 
-        await screen.findByRole("link", { name: "Take me to Metabase" });
+      const flags = await getLastSettingsPutPayload();
 
-        expect(localStorage.getItem("showEmbedHomepage")).toBeNull();
+      expect(flags).toMatchObject({
```

Comment:
I believe [`toMatchObject`](https://jestjs.io/docs/expect#tomatchobjectobject) does partial matching. You probably want to use `toEqual` in this case to ensure nothing else is in `flags`.

Reply:
Good advice, done in 882ed72b4a5583f2610ea2b683ee4309021cb52f